inductiva/tasks/task.py

```diff
@@ -172,13 +172,65 @@ def wait(self, polling_period: int = 5) -> models.TaskStatusCode:
 
             time.sleep(polling_period)
 
-    def kill(self) -> None:
-        """Kill the task.
-
-        This method issues a request to the API to kill the task. It doesn't
-        block waiting for confirmation if the task was killed.
+    def kill(self,
+             wait: bool = False,
+             timeout: float = 5,
+             maxretries: int = 5) -> bool:
+        """
+        Tries to kill a task. If wait is True (default is False)
+        the function sends the kill command and returns.
+        If wait is False the function checks for confirmation that
+        the kill command was successfull.
+        Args:
+            wait: Determines if we wait for the kill command to
+                finish.
+            timeout: Time in seconds to wait for the kill command.
+            maxretries: Number of times to retry the kill command.
+        Returns:
+            True if the kill command was successfull, False otherwise.
         """
-        self._api.kill_task(path_params=self._get_path_params())
+        while maxretries > 0:
+            maxretries -= 1
+            try:
+                self._api.kill_task(path_params=self._get_path_params())
+                break
+            except exceptions.ApiException as e:
+                logging.error("Error while sending the kill command: %s", e)
+                if maxretries == 0:
+                    logging.error(
+                        "Something went wrong while senting the kill"\
+                            "command. Max retries reached."
+                    )
+                    return False
+                logging.error(
+                    "Something went wrong while senting the kill command."\
+                        "Trying again..."
+                )
+
+        if not wait:
+            return True
+
+        success = True
+        start = time.time()
+        status = self.get_status()
+
+        while status == models.TaskStatusCode.PENDINGKILL:
+            if (time.time() - start) >= timeout:
+                success = False
+                logging.error("Kill command timed out.")
```

Comment:
Even if it times out, the status can still change to Killed. Which would raise an error below, since you don't update it again after the loop.

So the `self.get_status()` should be before the time check.